### PR TITLE
Add unit tests for BlockMasterSync

### PR DESCRIPTION
**Pull request by YangchenYe323**
### What changes are proposed in this pull request?
Added unit tests for `BlockMasterSync`

### Why are the changes needed?
To make sure that the syncing process is carried out correctly (master's command handled, errors processed, etc.)
Improved unit test coverage(0% -> 88%)

### Does this PR introduce any user facing changes?
No

### CODE REVIEW

**Review by YangchenYe323 — COMMENTED** [review]
This change might be discussed further. I changed `BlockMasterSync` to **read Configuration inline when needed** instead of **saving the configuration in static variables**. 

The reason is that static variables get finalized when the class is first loaded, but configuration might change after that first load. 

I think the expected behavior in that case is to make newly created instances of `BlockMasterSync` reflect new configurations.

File: core/server/worker/src/main/java/alluxio/worker/block/BlockMasterSync.java [hunk 1]
```diff
@@ -51,18 +51,6 @@
...
-  private static final long ACQUIRE_LEASE_WAIT_BASE_SLEEP_MS =
-      Configuration.getMs(PropertyKey.WORKER_REGISTER_LEASE_RETRY_SLEEP_MIN);
-  private static final long ACQUIRE_LEASE_WAIT_MAX_SLEEP_MS =
-      Configuration.getMs(PropertyKey.WORKER_REGISTER_LEASE_RETRY_SLEEP_MAX);
-  private static final long ACQUIRE_LEASE_WAIT_MAX_DURATION =
-      Configuration.getMs(PropertyKey.WORKER_REGISTER_LEASE_RETRY_MAX_DURATION);
-  private static final boolean USE_STREAMING =
-      Configuration.getBoolean(PropertyKey.WORKER_REGISTER_STREAM_ENABLED);
-  private static final int HEARTBEAT_TIMEOUT_MS =
-      (int) Configuration.getMs(PropertyKey.WORKER_BLOCK_HEARTBEAT_TIMEOUT_MS);
```
Comment (YangchenYe323) [inline core/server/worker/src/main/java/alluxio/worker/block/BlockMasterSync.java:65]: https://github.com/Alluxio/alluxio/pull/15731#pullrequestreview-1008162195

File: core/server/worker/src/test/java/alluxio/worker/block/BlockMasterSyncTest.java [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+  }
+
+  @Test
+  public void failToAcquireLease() throws Exception {
+    // simulates an error when acquiring lease
+    doThrow(new FailedToAcquireRegisterLeaseException("Failed Acquiring Lease"))
+        .when(mClient)
+        .acquireRegisterLeaseWithBackoff(
+            any(long.class),
+            any(int.class),
```
Comment (dbw9580) [inline core/server/worker/src/test/java/alluxio/worker/block/BlockMasterSyncTest.java:101]:
```suggestion
            anyLong(),
            anyInt(),
```

**Review by dbw9580 — COMMENTED** [review]
I'm afraid this feels like whitebox testing too much. This class should better be tested in an integration test instead, IMO. You can take a look at `alluxio.client.fs.BlockWorkerRegisterStreamIntegrationTest` to have an idea of how integration test works. Thanks for the PR, though.

**Comment by YangchenYe323** [comment]
> I'm afraid this feels like whitebox testing too much. This class should better be tested in an integration test instead, IMO. You can take a look at `alluxio.client.fs.BlockWorkerRegisterStreamIntegrationTest` to have an idea of how integration test works. Thanks for the PR, though.

I agree that this test looks like white box. The goal of this unit test is testing "the behavior of BlockMasterSync", rather than "BlockWorker and BlockMaster are in sync". 

The latter is covered in other intergation tests in which the whole class `BlockMasterSync` is transparent: they don't care what mechanism is used to sync up metadata, just that the metadata is in good state.
For example, `BlockMasterSync` already have 73% coverage in `BlockMasterIntegrityIntegrationTest`:
Screen Shot 2022-06-16 at 4 40 31 PM

I'm worried that the former cannot be tested "black-boxedly", as the behavior of `BlockMasterSync` is pretty much delegating work to its components.

**Comment by YangchenYe323** [comment]
I'll close this pr for now and do an integration test instead

**Pull request closed without merging**